src/components/AutoCompleteInput.tsx

```diff
@@ -0,0 +1,208 @@
+import {
+  Popover,
+  PopoverTrigger,
+  PopoverContent,
+  PopoverBody,
+  useColorModeValue,
+} from "@chakra-ui/react";
+import { Box, Text } from "@chakra-ui/react";
+import React, { useCallback, useEffect } from "react";
+import { useRef, useState } from "react";
+
+type AutoCompleteProps = {
+  children: React.ReactNode;
+  triggerRef: React.RefObject<HTMLTextAreaElement>;
+  parentRef: React.RefObject<HTMLDivElement>;
+  availablePrompts: {
+    command: string;
+    helpTitle: string;
+    helpDescription: string;
+  }[];
```

Comment:
Since this is a generic component, we should not couple the props with a specific use case like commands. For example, see how React MUI does this https://mui.com/material-ui/react-autocomplete/#options-structure

```suggestion
  options: {
    label: string;
  }[]
```

Its better to define an `AutoCompleteOption` type exclusively
```ts
type AutoCompleteOption = {
    label: string;
}
...
...
options: AutoCompleteOption[]
```

Reply:
I see we need to pass in all this complex option information and have custom styling, but the right way to do this is by specifying an option renderer function like so:
```ts
type AutoCompleteInputProps<OptionType> = {
  ...
  ...
  options: OptionType[];
  renderOption: (option: OptionType) => ReactNode;
};

function AutoCompleteInput<OptionType>(props: AutoCompleteInputProps<OptionType>) {
  return (
```

See https://mui.com/material-ui/react-autocomplete/#githubs-picker for reference.